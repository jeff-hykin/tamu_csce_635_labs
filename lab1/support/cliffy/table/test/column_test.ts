import { snapshotTest } from "../../testing/snapshot.ts";
import { Table } from "../table.ts";

const createTable = () =>
  new Table()
    .header(["Foo", "Bar", "Baz"])
    .body([
      ["foo bar baz", "baz", "beep boop"],
      ["baz", "beep boop", "foo bar baz"],
      ["beep boop", "foo bar baz", "baz"],
    ]);

await snapshotTest({
  name: "[table] should set border on columns",
  meta: import.meta,
  fn() {
    createTable()
      .columns([{
        border: true,
      }, {
        border: false,
      }, {
        border: true,
      }])
      .render();
  },
});

await snapshotTest({
  name: "[table] should set align on columns",
  meta: import.meta,
  fn() {
    createTable()
      .columns([{
        align: "left",
      }, {
        align: "center",
      }, {
        align: "right",
      }])
      .border()
      .render();
  },
});

await snapshotTest({
  name: "[table] should set width on columns",
  meta: import.meta,
  fn() {
    createTable()
      .columns([{
        maxWidth: 4,
      }, {
        minWidth: 20,
      }, {
        align: "right",
      }])
      .border()
      .render();
  },
});

await snapshotTest({
  name: "[table] should set padding on columns",
  meta: import.meta,
  fn() {
    createTable()
      .columns([{
        padding: 5,
        align: "left",
      }, {
        padding: 5,
        minWidth: 20,
        align: "center",
      }, {
        padding: 5,
        align: "right",
      }])
      .border()
      .render();
  },
});

await snapshotTest({
  name: "[table] should set column options with column method",
  meta: import.meta,
  fn() {
    const table = createTable();
    table.getColumn(0)?.padding(5);
    table.getColumn(0)?.align("left");

    table.getColumn(1)?.padding(5);
    table.getColumn(1)?.minWidth(20);
    table.getColumn(1)?.align("center");

    table.getColumn(2)?.padding(5);
    table.getColumn(2)?.align("right");

    table
      .border()
      .render();
  },
});
