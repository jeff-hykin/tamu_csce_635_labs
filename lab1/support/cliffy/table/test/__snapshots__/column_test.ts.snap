export const snapshot = {};

snapshot[`[table] should set border on columns 1`] = `
stdout:
"  Foo           Bar           Baz          
┌─────────────┐             ┌─────────────┐
│ foo bar baz │ baz         │ beep boop   │
├─────────────┤             ├─────────────┤
│ baz         │ beep boop   │ foo bar baz │
├─────────────┤             ├─────────────┤
│ beep boop   │ foo bar baz │ baz         │
└─────────────┘             └─────────────┘
"
stderr:
""
`;

snapshot[`[table] should set align on columns 1`] = `
stdout:
"┌─────────────┬─────────────┬─────────────┐
│ Foo         │ Bar         │ Baz         │
├─────────────┼─────────────┼─────────────┤
│ foo bar baz │     baz     │   beep boop │
├─────────────┼─────────────┼─────────────┤
│ baz         │  beep boop  │ foo bar baz │
├─────────────┼─────────────┼─────────────┤
│ beep boop   │ foo bar baz │         baz │
└─────────────┴─────────────┴─────────────┘
"
stderr:
""
`;

snapshot[`[table] should set width on columns 1`] = `
stdout:
"┌──────┬──────────────────────┬─────────────┐
│ Foo  │ Bar                  │ Baz         │
├──────┼──────────────────────┼─────────────┤
│ foo  │ baz                  │   beep boop │
│ bar  │                      │             │
│ baz  │                      │             │
├──────┼──────────────────────┼─────────────┤
│ baz  │ beep boop            │ foo bar baz │
├──────┼──────────────────────┼─────────────┤
│ beep │ foo bar baz          │         baz │
│ boop │                      │             │
└──────┴──────────────────────┴─────────────┘
"
stderr:
""
`;

snapshot[`[table] should set padding on columns 1`] = `
stdout:
"┌─────────────────────┬──────────────────────────────┬─────────────────────┐
│     Foo             │     Bar                      │     Baz             │
├─────────────────────┼──────────────────────────────┼─────────────────────┤
│     foo bar baz     │             baz              │       beep boop     │
├─────────────────────┼──────────────────────────────┼─────────────────────┤
│     baz             │          beep boop           │     foo bar baz     │
├─────────────────────┼──────────────────────────────┼─────────────────────┤
│     beep boop       │         foo bar baz          │             baz     │
└─────────────────────┴──────────────────────────────┴─────────────────────┘
"
stderr:
""
`;

snapshot[`[table] should set column options with column method 1`] = `
stdout:
"┌─────────────────────┬──────────────────────────────┬─────────────────────┐
│     Foo             │     Bar                      │     Baz             │
├─────────────────────┼──────────────────────────────┼─────────────────────┤
│     foo bar baz     │             baz              │       beep boop     │
├─────────────────────┼──────────────────────────────┼─────────────────────┤
│     baz             │          beep boop           │     foo bar baz     │
├─────────────────────┼──────────────────────────────┼─────────────────────┤
│     beep boop       │         foo bar baz          │             baz     │
└─────────────────────┴──────────────────────────────┴─────────────────────┘
"
stderr:
""
`;
